api/alerts.go

```diff
@@ -108,14 +108,13 @@ func (a *Alerts) Update(viewName string, newAlert *Alert) (*Alert, error) {
 	return &alert, nil
 }
 
-func (a *Alerts) Add(viewName string, newAlert *Alert) (*Alert, error) {
+func (a *Alerts) Add(viewName string, newAlert *Alert, runAsUserID string) error {
```

Comment:
Is it possible to represent no run-as-user as something other than an empty string?

Reply:
The default value is set to "", if not specified, and is as such "optional"

`// StringVar defines a string flag with specified name, default value, and usage string.
// The argument p points to a string variable in which to store the value of the flag.
func (f *FlagSet) StringVar(p *string, name string, value string, usage string) {
	f.VarP(newStringValue(value, p), name, "", usage)
}`

Reply:
@jespermbyskov There are no "Option" types in Go. The closest you can get, is a pointer to a value, which can be nil. Though that is/can be ugly/unnatural to use.

In this case, the added parameter is also a breaking change of a public API. There is only a single way to solve this, without creating duplicate methods, which is to use variable argument lists. Adding a final varargs parameter would be backwards compatible.

@fjerlov-cs The way to do this, would be to change the method to be:

```suggestion
type AlertsAddOpts struct {
  runAsUserID string
}

type AlertsAddOpt func(*AlertsAddOpts)

func AlertsAddRunAsUserID(v string) AlertsAddOpt {
  return func (o *AlertsAddOpts) {
    o.runAsUserID = v
  }
}

func (a *Alerts) Add(viewName string, newAlert *Alert, optsFn ...AlertsAddOpt) (*Alert, error) {
  var opts AlertsAddOpts
  for _, fn := range optsFn {
    fn(&opts)
  }
  
  // the optional argument can be accessed through:
  // runAsUserID := opts.runAsUserID
```

On the call site the it would be:

```go
// var alerts *Alerts

alerts.Add("viewName", &Alert{})
// or with run as user id:
alerts.Add("viewName", &Alert{}, AlertsAddRunAsUserID("userID"))
```

Of course, it's also possible to distinguish between a present vs non-present optional argument in this case, by changing `runAsUserID string` in the above struct to `*string` or adding a bool that is set to true, if the argument was present. Whether that is necessary, I would say really depends on the meaning of an empty string as an argument.

Reply:
Actually missed an important point though. Why isn't the `RunAsUserID` just another field on the `Alert` struct, like the other fields for creating the alert?

Reply:
After a talk with @danielamkaer. I believe the best approach is to include it in `Alert struct` instead. Although is it a bit unintuitive with the `Alert -> CreateAlert -> Alert` struct flow, it will keep original pattern for creating and exporting alerts. 

It is now an optional in our alert.yaml for install, and forced on export.

If present upon creation, it will set the runAsUser when creating the alert.

```
runAsUser:
  id: XXXXX
```

The cli have no implementation for updating the alert, e.g. using `(a *Alerts) Update(viewName string, newAlert *Alert)`, however I will add that in an upcoming PR.